### PR TITLE
feat: Add support for bearer/auth token request

### PR DESCRIPTION
**Pull request by amopatel1234**
# DCMAW-6241
_Thank you for your contribution to the project._

Added support for authentication network request and updated docs to reflect new changes

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [ ] ~Created a `draft` pull request if it is not yet ready for review~

## Before your pull request can be reviewed:
- [ ] ~Met all of the acceptance criteria specified in the user story on Jira~
- [x] Reviewed your own code to ensure you are following the style guidelines
- [ ] ~Ran the app and tested the feature on a range of device sizes~
      ~Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.~
- [x] Written Unit and Integration tests if needed

- [ ] ~Met all accessibility requirements?~
    - [ ] ~Checked dynamic type sizes are applied~
    - [ ] ~Checked VoiceOver can navigate your new code~
    - [ ] ~Checked a user can navigate only using a keyboard around your new code~

## Before merging your pull request:
- [ ] Ensure that the code coverage and SonarCloud checks have passed
- [ ] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [ ] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Tests/NetworkingTests/NetworkClientTests.swift [hunk 1]
```diff
@@ -52,4 +52,36 @@ extension NetworkClientTests {
             XCTAssert(error is ServerError)
         }
     }
+    
+    func test_bearerToken_isInRequest() async throws {
+        sut = nil
+        sut = .init(configuration: configuration, authenticationProvider: MockAuthenticationProvider())
```
Comment (Oliver-Binns) [inline Tests/NetworkingTests/NetworkClientTests.swift:58]: I would move this sut setup into the test setup above

File: Tests/NetworkingTests/NetworkClientTests.swift [hunk 2]
```diff
@@ -52,4 +52,36 @@ extension NetworkClientTests {
...
+        }
+        
+        let bearerToken = request.value(forHTTPHeaderField: "Authorization")
+        // swiftlint:disable line_length
+        let tokenToCheck = """
+      Bearer zh2JZEiVWK9AHJNQN3TiK7MFjbuECjyA-I9uL9oFHmNZv0v4HnJrs64ZiNjp9nOSJjAM7gBIWJXlnzxG07/wzSE?neXnLinU6YYAZ!Q2CRwxkeqWT=hBapLWnbP1g/5j2MXlnY7gw2T=?i6nBI86=MJ1NR!fxqlwg4iI73x9HTL15i6urRrICHEps2BEy0c0uAet6USg6dpy0fFnf6Cf!ZozO37TOJB=T7!FooE8HDlM8WXggsO6cDQdDzRS43gL
+      """
+        // swiftlint:enable line_length
+        
+        XCTAssertEqual(bearerToken, tokenToCheck)
```
Comment (Oliver-Binns) [inline Tests/NetworkingTests/NetworkClientTests.swift:85]: Do we want another test to check the absence of a bearer token if no authentication provider is injected?

**Review by Oliver-Binns — APPROVED** [review]
Happy overall. I've left two queries but they're not dealbreakers for me.

File: Tests/NetworkingTests/NetworkClientTests.swift (hunk 2 above)
Comment (amopatel1234) [reply Tests/NetworkingTests/NetworkClientTests.swift:85]: will add this in a later pr

**Pull request merged**